src/sbt-test/sbt-scalafix/scalafixOnCompileConfig/test

```diff
@@ -0,0 +1,37 @@
+# check if scalafix-on-compile is disabled
+
+> example/compile
+-> example/scalafix
+
+# check if scalafix-on-compile is enabled and uses .scalafix.conf
+
+> set scalafixOnCompile.in(ThisBuild) := true
+-> example/compile
+-> example/scalafix
+
+# check if scalafix-on-compile uses .scalafix-on-compile.conf
+#   and scalafix-explicit-run still uses .scalafix.conf
+
+> set scalafixOnCompileConfig.in(ThisBuild) := file(".scalafix-on-compile.conf"),
+-> example/scalafix
+-> example/scalafix --check RemoveUnused
+> example/compile
+## A statement incl. null has been removed by the precedence scalafix-on-compile.
+> example/scalafix
```

Comment:
to actually assert on the rewrite, you can use `newer`, see https://github.com/scalacenter/sbt-scalafix/blob/11d4c3f1bb182c2eb8c7b81ebcb6535f85a3f98e/src/sbt-test/skip-windows/caching/test#L42-L45

Reply:
In this scripted test, `> example/scalafix` always runs only DisableSyntax, so that it won't rewrite anything.

Only `example/compile` or `nonScalafixConfig/compile` may rewrite. So I'll add some assertion about their rewrites.